fluid/object_detection/eval.py

```diff
@@ -74,33 +66,123 @@ def if_exist(var):
+                    for j in range(dt_num_this_img):
+                        dt = nmsed_out_v[k]
+                        k = k + 1
+                        category_id, score, xmin, ymin, xmax, ymax = dt.tolist()
+                        xmin = max(min(xmin, 1.0), 0.0) * image_width
+                        ymin = max(min(ymin, 1.0), 0.0) * image_height
+                        xmax = max(min(xmax, 1.0), 0.0) * image_width
+                        ymax = max(min(ymax, 1.0), 0.0) * image_height
+                        w = xmax - xmin
+                        h = ymax - ymin
+                        bbox = [xmin, ymin, w, h]
+                        dt_res = {
+                            'image_id': image_id,
+                            'category_id': category_id,
+                            'bbox': bbox,
+                            'score': score
+                        }
+                        dts_res.append(dt_res)
+
+            with open("detection_result.json", 'w') as outfile:
+                json.dump(dts_res, outfile)
+            print("start evaluate using coco api")
+            from pycocotools.coco import COCO
+            from pycocotools.cocoeval import COCOeval
+            cocoGt = COCO(os.path.join(data_args.data_dir, test_list))
+            cocoDt = cocoGt.loadRes("detection_result.json")
+            cocoEval = COCOeval(cocoGt, cocoDt, "bbox")
+            cocoEval.evaluate()
+            cocoEval.accumulate()
+            cocoEval.summarize()
 
-    _, accum_map = map_eval.get_map_var()
-    map_eval.reset(exe)
-    for idx, data in enumerate(test_reader()):
-        test_map = exe.run(test_program,
-                           feed=feeder.feed(data),
-                           fetch_list=[accum_map])
-        if idx % 50 == 0:
-            print("Batch {0}, map {1}".format(idx, test_map[0]))
-    print("Test model {0}, map {1}".format(model_dir, test_map[0]))
+        else:
+            test_program = fluid.default_main_program().clone(for_test=True)
+            with fluid.program_guard(test_program):
+                map_eval = fluid.evaluator.DetectionMAP(
+                    nmsed_out,
+                    gt_label,
+                    gt_box,
+                    difficult,
+                    num_classes,
+                    overlap_threshold=0.5,
+                    evaluate_difficult=False,
+                    ap_version=args.ap_version)
+
+            _, accum_map = map_eval.get_map_var()
+            map_eval.reset(exe)
+            for batch_id, data in enumerate(test_reader()):
+                test_map = exe.run(test_program,
+                                   feed=feeder.feed(data),
+                                   fetch_list=[accum_map])
+                if batch_id % 20 == 0:
+                    print("Batch {0}, map {1}".format(batch_id, test_map[0]))
+            print("Test model {0}, map {1}".format(model_dir, test_map[0]))
+
+    test()
```

Comment:
coco could be evaluated in 11point, integral and cocoMAP, while pascalvoc is just allowed be evaluted in 11point and integral. So I separate the cocoMAP and [11point, integral] into two files rather than coco and pascalvoc.